metaflow/plugins/aws/awslambda/lambda_decorator.py

```diff
@@ -0,0 +1,208 @@
+    headline = "AWS Lambda error"
+
+
+class LambdaDecorator(StepDecorator):
+    """
+    The decorator is called @awslambda to avoid confusion, since "lambda" is a reserved
+    word in Python.
+    """
+
+    name = "awslambda"
+    defaults = {
+        "image": None,
+        "force_fetch_image": False,
+    }
+
+    package_url = None  # type: Optional[str]
+    package_sha = None  # type: Optional[str]
+    resource_attributes = {}  # type: Dict[str, str]
+    run_time_limit = None
+    ds_root = None  # type: Optional[str]
+
+    def step_init(
+        self,
+        flow,  # type: "FlowSpec"
+        graph,  # type : "FlowGraph"
+        step,  # type: str
+        decos,  # type: List[Decorator]
+        environment,  # type: "MetaflowEnvironment"
+        datastore,  # type: "MetaflowDataStore"
+        logger,  # type: "EventLogger"
+    ):
+        if datastore.TYPE != "s3":
+            raise LambdaException("The *@awslambda* decorator requires --datastore=s3.")
+
+        self.logger = logger
+        self.environment = environment
+        self.step = step
+        for deco in decos:
+            if isinstance(deco, CondaStepDecorator):
+                raise LambdaException(
+                    "*@awslambda* doesn't support using @conda. "
+                    "We recommend using custom docker images to manage"
+                    " dependencies instead"
+                )
+
+            if isinstance(deco, BatchDecorator):
+                raise LambdaException(
+                    "You can't use @batch and *@awslambda* at the same time"
+                )
+
+            if isinstance(deco, ResourcesDecorator):
+                for k, v in deco.attributes.items():
+                    if v != deco.defaults[k]:
+                        if k != "memory":
+                            raise LambdaException(
+                                "*@awslambda* only supports provisioning memory via @resources, not {r}".format(
+                                    r=k
+                                )
+                            )
+                        else:
+                            if float(v) > LAMBDA_MAX_MEMORY_MB:
+                                raise LambdaException(
+                                    "{r} cannot exceed {max_v} for *@awslambda* (currently set to {v})".format(
+                                        r=k, max_v=LAMBDA_MAX_MEMORY_MB, v=v
+                                    )
+                                )
+                        self.resource_attributes[k] = str(v)
+        self.run_time_limit = get_run_time_limit_for_task(
+            decos, default_limit=DEFAULT_TIMEOUT_SECONDS
+        )
+        if self.run_time_limit < 10:
+            raise LambdaException(
+                "The timeout for step *{step}* should be at "
+                "least 10 seconds for execution on AWS Lambda".format(step=step)
+            )
+        if self.run_time_limit > 900:
```

Comment:
Ditto as the above comment.

Reply:
k will ignore it if the limit it too high to be consistent